src/garage/torch/baselines/gaussian_mlp_baseline.py

```diff
@@ -0,0 +1,170 @@
+                 learn_std=True,
+                 init_std=1.0,
+                 min_std=1e-6,
+                 max_std=None,
+                 std_parameterization='exp',
+                 layer_normalization=False,
+                 normalize_observation=True,
+                 normalize_return=True,
+                 optimizer=torch.optim.Adam,
+                 optimizer_args=None):
+        super().__init__(env_spec)
+
+        self._module = GaussianMLPModule(
+            input_dim=env_spec.observation_space.flat_dim,
+            output_dim=1,
+            hidden_sizes=hidden_sizes,
+            hidden_nonlinearity=hidden_nonlinearity,
+            hidden_w_init=hidden_w_init,
+            hidden_b_init=hidden_b_init,
+            output_nonlinearity=output_nonlinearity,
+            output_w_init=output_w_init,
+            output_b_init=output_b_init,
+            learn_std=learn_std,
+            init_std=init_std,
+            min_std=min_std,
+            max_std=max_std,
+            std_parameterization=std_parameterization,
+            layer_normalization=layer_normalization)
+
+        self._normalize_observation = normalize_observation
+        self._normalize_return = normalize_return
+
+        if optimizer_args is None:
+            optimizer_args = dict()
+
+        self._optimizer = make_optimizer(optimizer, self._module,
+                                         **optimizer_args)
+
+    def fit(self, paths):
+        """Fit regressor based on paths.
+
+        Args:
+            paths (list[dict]): Sample paths.
+
+        """
+        # observations = xs, return = ys
+        observations = np.concatenate([p['observations'] for p in paths])
+        returns = np.concatenate([p['returns'] for p in paths])
+
+        if self._normalize_observation:
+            mean = np.mean(observations, axis=0, keepdims=True)
+            std = np.std(observations, axis=0, keepdims=True) + 1e-8
+            observations = (observations - mean) / std
+
+        if self._normalize_return:
+            mean = np.mean(returns, axis=0, keepdims=True)
+            std = np.std(returns, axis=0, keepdims=True) + 1e-8
+            returns = (returns - mean) / std
+
+        dist = self._module(torch.Tensor(observations))
+        ll = dist.log_prob(torch.Tensor(returns))
+        loss = -ll.mean()
+
+        self._optimizer.zero_grad()
+        loss.backward()
+        self._optimizer.step()
+
+    def predict(self, path):
+        """Predict value based on paths.
+
+        Args:
+            path (list[dict]): Sample paths.
+
+        Returns:
+            numpy.ndarray: Predicted value.
+
+        """
+        with torch.no_grad():
+            dist = self._module(torch.Tensor(path['observations']))
+        return dist.mean.flatten().numpy()
+
+    def get_param_values(self):
+        """Get the parameters to the policy.
+
+        This method is included to ensure consistency with TF policies.
```

Comment:
why? is this used by the torch algorithms?

Reply:
Sorry, this is a copy and paste mistake.